src/trackinfoobject.cpp

```diff
@@ -357,64 +360,65 @@ bool TrackInfoObject::exists() const {
     return QFile::exists(m_fileInfo.absoluteFilePath());
 }
 
-float TrackInfoObject::getReplayGain() const {
+double TrackInfoObject::getReplayGain() const {
     QMutexLocker lock(&m_qMutex);
     return m_fReplayGain;
 }
 
-void TrackInfoObject::setReplayGain(float f) {
+void TrackInfoObject::setReplayGain(double replayGain) {
     QMutexLocker lock(&m_qMutex);
-    //qDebug() << "Reported ReplayGain value: " << m_fReplayGain;
-    if (m_fReplayGain != f) {
-        m_fReplayGain = f;
-        setDirty(true);
+    if (modifyFieldValue(&m_fReplayGain, replayGain)) {
+        setDirtyAndUnlock(&lock, true);
+        emit(ReplayGainUpdated(replayGain));
     }
-    lock.unlock();
-    emit(ReplayGainUpdated(f));
 }
 
 double TrackInfoObject::getBpm() const {
     QMutexLocker lock(&m_qMutex);
-    if (!m_pBeats) {
-        return 0;
-    }
-    // getBpm() returns -1 when invalid.
-    double bpm = m_pBeats->getBpm();
-    if (bpm >= 0.0) {
-        return bpm;
+    return getBpmLocked();
+}
+
+double TrackInfoObject::getBpmLocked() const {
```

Comment:
The locked prefix is misleading since it dos not read locked, it requires locked. 
What about avoid it and use "getBpmInner()"